src/SFA.DAS.EmployerAccounts/Commands/CreateUserAccount/CreateUserAccountCommandHandler.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.Audit.Types;
+using SFA.DAS.Authorization;
+using SFA.DAS.EmployerAccounts.Commands.AuditCommand;
+using SFA.DAS.EmployerAccounts.Commands.PublishGenericEvent;
+using SFA.DAS.EmployerAccounts.Data;
+using SFA.DAS.EmployerAccounts.Factories;
+using SFA.DAS.EmployerAccounts.Messages.Events;
+using SFA.DAS.EmployerAccounts.Models;
+using SFA.DAS.EmployerAccounts.Models.Account;
+using SFA.DAS.EmployerAccounts.Models.UserProfile;
+using SFA.DAS.EmployerAccounts.Queries.GetUserByRef;
+using SFA.DAS.Hashing;
+using SFA.DAS.HashingService;
+using SFA.DAS.NServiceBus;
+using SFA.DAS.Validation;
+using Entity = SFA.DAS.Audit.Types.Entity;
+
+
+namespace SFA.DAS.EmployerAccounts.Commands.CreateUserAccount
+{
+    //TODO this needs changing to be a facade and calling individual commands for each component
+    public class CreateUserAccountCommandHandler : IAsyncRequestHandler<CreateUserAccountCommand, CreateUserAccountCommandResponse>
```

Comment:
There is something I don't like about this class and it needs a significant refactor (it's doing way too much).  I realise it was a copy from somewhere else and now probably isn't the appropriate time to resolve it but it's worth having a chat about on Monday maybe?

Reply:
@MichaelYoung1981  Dan's  //TODO comment at the top of the originating class is from early 2017, so definitely overdue a refactor :0)